tests/src/test/scala/com/nvidia/spark/rapids/HashAggregatesSuite.scala

```diff
@@ -1566,14 +1568,19 @@ class HashAggregatesSuite extends SparkQueryCompareTestSuite {
       res
   }
 
-  // A test that verifies that Distinct with Filter is not supported on the CPU or the GPU.
-  testExpectedExceptionStartsWith(
-      "Avg Distinct with filter - unsupported on CPU and GPU",
-      classOf[AnalysisException],
-      "DISTINCT and FILTER cannot be used in aggregate functions at the same time",
-      longsFromCSVDf, conf = floatAggConf) {
-    frame => val res = frame.selectExpr("avg(distinct longs) filter (where longs < 5)")
-      res
+  if (spark.SPARK_VERSION_SHORT < "3.1.0") {
```

Comment:
this is fine but we had added a function to the shim layer to get the version so that we could handle different shim layers in addition to apache versions.  There is one test using it already, it would be nice to stay consistent (see StringFallbackSuite):

      val isValidTestForSparkVersion = ShimLoader.getSparkShims.getSparkShimVersion match {
        case SparkShimVersion(major, minor, _) => major == 3 && minor == 0
        case DatabricksShimVersion(major, minor, _) => major == 3 && minor == 0
        case _ => true
      }